services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbAsyncTable.java

```diff
@@ -667,4 +667,28 @@
     default CompletableFuture<T> updateItem(T item) {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * Deletes a table in DynamoDb with the name and schema already defined for this DynamoDbTable.
+     * <p>
+     * Use {@link DynamoDbEnhancedClient#table(String, TableSchema)} to define the mapped table resource.
+     * <p>
+     * This operation calls the low-level DynamoDB API DeleteTable operation. Note that this is an asynchronous operation and that
+     * the table may not immediately deleted. Currently, there is no mechanism supported within this
+     * library to wait for/check the status of a deleted table. You must provide this functionality yourself.
```

Comment:
They can use the waiters in the underlying client to do it.

Reply:
While we are at it, should we add `deleteTableUntilTableNotExists` instead of `deleteTable` here?

Reply:
We have `waitUntilTableNotExists` in the `DynamoDbWaiter`.

If we decide to use the waiter in this operation, we can name it `deleteTableAndWaitUntilTableNotExists` or something similar.

Reply:
I'd personally rather just have deleteTable and have them call the waiter afterwards if they want to wait.

Reply:
Agree. Updated the comment , directing them to use waitUntilTableNotExists in the Waiters.

Reply:
Yeah, good point. People might be more interested to wait for a table to be created than deleted.